argilla/src/argilla/records/_dataset_records.py

```diff
@@ -398,92 +410,103 @@ def _validate_vector_names(self, vector_names: Union[List[str], str]) -> None:
             if vector_name not in self.__dataset.schema:
                 raise ValueError(f"Vector field {vector_name} not found in dataset schema.")
 
+    def _render_record_mapping(
+        self,
+        records: List[Dict[str, Any]],
+        mapping: Optional[Dict[str, Union[str, List[str], Tuple[str]]]] = None,
+    ) -> Dict[str, Tuple[Optional[str]]]:
+        """Renders a mapping from a list of records and a mapping dictionary, to a singular mapping dictionary."""
+        schema = self.__dataset.schema
+        mapping = mapping or {}
+        singular_mapping = defaultdict(dict)
+
+        # update the mapping with unmapped columns
+        for key, value in records[0].items():
+            if key not in schema and key not in mapping:
+                warnings.warn(
+                    message=f"Record attribute {key} is not in the schema so skipping.",
+                )
+            if key not in mapping:
+                mapping[key] = key
+
+        # create a singular mapping with destinations from the schema
+        for source_key, value in mapping.items():
+            destinations = [value] if isinstance(value, str) else list(value)
+
+            for attribute_mapping in destinations:
+                attribute_mapping = attribute_mapping.split(".")
+
+                attribute_name = attribute_mapping[0]
+                schema_item = schema.get(attribute_name)
+                attribute_type = attribute_mapping[1] if len(attribute_mapping) > 1 else None
```

Comment:
don't we want to check if all attribute types are the same and if not can't there be different attribute types? perhaps I want to map to metadata and field or something?

Reply:
Sorry, I don't understand this question. The user can map a value in their data by doing:

```python
mapping = {
        "my_prompt": ("prompt_field", "prompt_question"),
    }
```

The strings in the tuple could point to metadata etc.